### PR TITLE
[13.0] [MIG] fieldservice_skill

### PR DESCRIPTION
**Pull request by brian10048**
Migrate fieldservice_skill to 13.0

For #354

### CODE REVIEW

**Comment by bodedra** [comment]
This PR depends on "hr_skill" module and its functionality come in Odoo13 hr_skills module.

Firefox_Screenshot_2020-07-21T19-02-24 423Z

So change module dependency from "hr_skill" to "hr_skills"

File: fieldservice_skill/__manifest__.py
Comment (bodedra) [inline fieldservice_skill/__manifest__.py]:
```suggestion
    "depends": ["hr_skills", "fieldservice"],
```

**Review by bodedra — CHANGES_REQUESTED** [review]

**Comment by brian10048** [comment]
> 
> 
> This PR depends on "hr_skill" module and its functionality come in Odoo13 hr_skills module.
> 
> Firefox_Screenshot_2020-07-21T19-02-24 423Z
> 
> So change module dependency from "hr_skill" to "hr_skills"

I updated the dependency and updated an inherited view name.

Will a migration script be needed because of this module change?

**Comment by bodedra** [comment]
@brian10048 I don't think we need a migration script because it has same model name "hr.skill" So we are safe.

cc @max3903 @dreispt

**Comment by RLeeGML** [comment]
@max3903 @osi-scampbell 

Closed my PR, I missed that this was already open. Same questions as before.

The dependency hr_skill from OCA/hr has been changed to hr_skills from odoo/odoo. The old module had a color field on hr.skill model, while the new one does not. Does this need to be added back in?

The Usage file indicates that the worker list on fsm.order will be filtered by worker skill (as well as location) but there's no functionality to filter by worker skill in the v12 module. Should this be added in?

File: fieldservice_skill/views/hr_skill.xml
Comment (murtuzasaleh) [inline fieldservice_skill/views/hr_skill.xml]: @brian10048 Can you please improve the action name or create new action of skills and bind it here?

**Comment by dreispt** [comment]
🚨 Red build!

**Comment by brian10048** [comment]
@RLeeOSI I saw your PR at [https://github.com/brian10048/field-service/pull/16](https://github.com/brian10048/field-service/pull/16), but I wonder if we should modify the v12 behavior to instead open the Skill Type view as it provides a more complete overview of default skills configuration like I recently updated here.

After looking at this more, I wonder if we should update the `fsm.person.skill` to more or less mimic [`hr.employee.skill`](https://github.com/odoo/odoo/blob/e9c54f6b3d5c36b658308e1900ab68dcc9de0357/addons/hr_skills/models/hr_skills.py#L16:30)
The Odoo skill module has a level progress feature that can replace the level priority style field on fsm.person.skill

I don't mind making the changes, just want to make sure we are all on the same page first.  @dreispt @murtuzasaleh  Let me know what you guys think as well

**Comment by RLeeGML** [comment]
@brian10048 I agree, it looks like the Odoo module intends to use hr.skill.type as the main model for skills (hence why there's a menuitem for it and not hr.skill). @max3903 and @osi-scampbell your input here would be greatly appreciated, and on the earlier comment [here](https://github.com/OCA/field-service/pull/602#issuecomment-706433962)

**Comment by max3903** [comment]
> The dependency hr_skill from OCA/hr has been changed to hr_skills from odoo/odoo. The old module had a color field on hr.skill model, while the new one does not. Does this need to be added back in?

Yes

> The Usage file indicates that the worker list on fsm.order will be filtered by worker skill (as well as location) but there's no functionality to filter by worker skill in the v12 module. Should this be added in?

Yes

**Comment by dreispt** [comment]
@brian10048 Lint checks are failing.
Can you check if you have [pre-commit configured](https://github.com/OCA/maintainer-quality-tools/#qa-tools-for-odoo-maintainers-mqt) on your local v13 repository?

**Comment by brian10048** [comment]
@RLeeOSI - to continue the conversation here for the others to see...
> @brian10048 do we need the migration script? The module name changes but the model is still "hr.skill"

Good catch.  The Odoo module and file name says hr_skills, but the model name is hr.skill like it was for the OCA - So this particular migration script is not relevant.  I think one would be required for those migrating from the OCA skills module to Odoo skills module, but that is probably not in the scope of field-service ??

Also, the domain filter I have added on the orders view to filter workers by skill is not working correctly.  I need to fix this as well and will fix those lint errors

**Comment by bodedra** [comment]
@brian10048 I agree with you.

There is no need migration script or any special treatment. fieldservice_skill will depends on Odoo's hr_skills module.

Would you please fix linting erros.

File: fieldservice_skill/views/fsm_order.xml
Comment (bodedra) [inline fieldservice_skill/views/fsm_order.xml]:
```suggestion
```

File: fieldservice_skill/views/fsm_order.xml
Comment (bodedra) [inline fieldservice_skill/views/fsm_order.xml:14]:
```suggestion
                />
                <field name="skill_worker_ids" invisible="1" />
```

**Review by bodedra — COMMENTED** [review]
minor one

**Comment by brian10048** [comment]
image

**Review by bodedra — APPROVED** [review]
Functional reviewed. :+1: 

Non-blocking comment.

I have found one minor gap.

When we set/update `Required Skills`, `Assigned To` list does not filter until save record.

Filter logic has implemented on compute field. So if we add onchange on `Required Skills` with same logic, `Assigned To` list will be filter without save record.

**Review by bodedra — APPROVED** [review]
Awesome. LG :+1:

**Comment by brian10048** [comment]
Screenshot from 2020-10-28 16-37-08

I am implementing a cleaner look for the worker skills to match the Odoo v13 skills look for employees.

I will have that ready for a final review soon

**Comment by bodedra** [comment]
@brian10048 Would you please fix this?

Screenshot from 2020-10-29 16-17-56

**Comment by brian10048** [comment]
> 
> 
> @brian10048 Would you please fix this?
>

Ready for final review @bodedra

**Review by bodedra — APPROVED** [review]
@brian10048 Functional reviewed. works great! :+1: 

+1 for beautiful view.

Screenshot from 2020-10-30 11-06-44

**Comment by bodedra** [comment]
@max3903 @dreispt would you please give your blessing to this PR!

**Comment by brian10048** [comment]
Thanks @bodedra, we can thank Odoo for the design of the view. I just thought it should remain cohesive here!

**Review by marcelsavegnago — APPROVED** [review]
LGTM. It's really pretty. Congratulations.

**Review by max3903 — APPROVED** [review]

**Pull request merged**
